### PR TITLE
Add nondeterministic alert to CUDA cumsum

### PR DESCRIPTION
**Pull request by kurtamohler**
Part of #75240

### CODE REVIEW

File: aten/src/ATen/native/cuda/ScanKernels.cpp [hunk 1]
```diff
@@ -88,6 +88,9 @@ Tensor _logcumsumexp_cuda(const Tensor& self, int64_t dim) {
 }
 
 void cumsum_cuda_kernel(const Tensor& result, const Tensor& self, int64_t dim) {
+  // See Note [Writing Nondeterministic Operations]
+  // Issue reporting nondeterministic behavior: https://github.com/pytorch/pytorch/issues/75240
+  globalContext().alertNotDeterministic("cumsum_cuda_kernel");
```
Comment (ngimel) [inline aten/src/ATen/native/cuda/ScanKernels.cpp:93]: Hm, if we don't want to break the world (a lot of workloads enable deterministic flag and do cumsums) we'll have to limit it to floating datatypes only.

Comment (kurtamohler) [reply aten/src/ATen/native/cuda/ScanKernels.cpp]: Changed to only alert for floating and complex

**Comment by ngimel** [comment]
@kurtamohler can you please rebase this?

**Comment by kurtamohler** [comment]
@ngimel, done

**Review by ngimel — APPROVED** [review]

**Pull request closed without merging**